### ISSUE TITLE
apiservice status False

### ISSUE THREAD

**Issue by mangareader** [kedacore/keda#1315]
Install via yaml file https://github.com/kedacore/keda/releases/download/v2.0.0/keda-2.0.0.yaml
and get error 
```
kubectl describe apiservice v1beta1.external.metrics.k8s.io

API Version:  apiregistration.k8s.io/v1
Kind:         APIService
Metadata:
  Creation Timestamp:  2020-11-05T03:56:36Z
  Resource Version:    358908819
  Self Link:           /apis/apiregistration.k8s.io/v1/apiservices/v1beta1.external.metrics.k8s.io
  UID:                 cc54bed8-55e2-48cf-9126-55db79597288
Spec:
  Group:                     external.metrics.k8s.io
  Group Priority Minimum:    100
  Insecure Skip TLS Verify:  true
  Service:
    Name:            keda-metrics-apiserver
    Namespace:       keda
    Port:            443
  Version:           v1beta1
  Version Priority:  100
Status:
  Conditions:
    Last Transition Time:  2020-11-05T03:56:36Z
    Message:               failing or missing response from https://10.44.12.6:6443/apis/external.metrics.k8s.io/v1beta1: Get https://10.44.12.6:6443/apis/external.metrics.k8s.io/v1beta1: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
    Reason:                FailedDiscoveryCheck
    Status:                False
    Type:                  Available
Events:                    <none>
```

## Specifications

  - **KEDA Version:** *v2*
  - **Platform & Version:** *GKE*
  - **Kubernetes Version:** *1.16.15-gke.3500*
  - **Scaler(s):** *Please elaborate*

**Comment by zroubalik** [comment]
Hi, this is a known issue of GKE , see our troubleshooting guide: https://keda.sh/docs/2.0/troubleshooting/

**Comment by mangareader** [comment]
Clusters with version 1.16.13+ will also have the fix.
Our gke version is  1.16.15

**Comment by zroubalik** [comment]
@mangareader are both KEDA pods running correctly? Anything suspicious in the logs?

**Comment by mangareader** [comment]
```
k get pod -n keda
NAME                                      READY   STATUS    RESTARTS   AGE
keda-metrics-apiserver-5f559d77d5-hbrj9   1/1     Running   0          6h18m
keda-operator-86dfcdf979-mh5w6            1/1     Running   0          6h18m
```

```
k logs -f keda-metrics-apiserver-65c46b9b8c-pn9mk -n keda
I1105 10:17:31.021578       1 main.go:83] keda_metrics_adapter "msg"="KEDA Version: 2.0.0"
I1105 10:17:31.021643       1 main.go:84] keda_metrics_adapter "msg"="KEDA Commit: 50bec808f8b4656922797e4bcb7c2b61fabd6d0e"
I1105 10:17:31.021652       1 main.go:85] keda_metrics_adapter "msg"="Go Version: go1.15.3"
I1105 10:17:31.021659       1 main.go:86] keda_metrics_adapter "msg"="Go OS/Arch: linux/amd64"
I1105 10:17:33.072896       1 request.go:621] Throttling request took 1.047798907s, request: GET:https://10.108.0.1:443/apis/monitoring.coreos.com/v1?timeout=32s
I1105 10:17:34.326681       1 provider.go:46] keda_metrics_adapter/provider "msg"="starting"
I1105 10:17:34.326715       1 main.go:114] keda_metrics_adapter "msg"="starting adapter..."
2020-11-05 10:17:34.326789 I | Starting metrics server at :9022
I1105 10:17:34.916012       1 serving.go:307] Generated self-signed cert (apiserver.local.config/certificates/apiserver.crt, apiserver.local.config/certificates/apiserver.key)
I1105 10:17:35.433420       1 configmap_cafile_content.go:202] Starting client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file
I1105 10:17:35.433458       1 shared_informer.go:223] Waiting for caches to sync for client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file
I1105 10:17:35.433483       1 configmap_cafile_content.go:202] Starting client-ca::kube-system::extension-apiserver-authentication::client-ca-file
I1105 10:17:35.433499       1 shared_informer.go:223] Waiting for caches to sync for client-ca::kube-system::extension-apiserver-authentication::client-ca-file
I1105 10:17:35.433818       1 dynamic_serving_content.go:130] Starting serving-cert::apiserver.local.config/certificates/apiserver.crt::apiserver.local.config/certificates/apiserver.key
I1105 10:17:35.434244       1 secure_serving.go:178] Serving securely on [::]:6443
I1105 10:17:35.434285       1 tlsconfig.go:240] Starting DynamicServingCertificateController
I1105 10:17:35.533637       1 shared_informer.go:230] Caches are synced for client-ca::kube-system::extension-apiserver-authentication::client-ca-file
I1105 10:17:35.533771       1 shared_informer.go:230] Caches are synced for client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file
```

```
k logs -f keda-operator-64657f8ccb-8nxtk -n keda
I1105 10:17:31.632168       1 request.go:621] Throttling request took 1.047887298s, request: GET:https://10.108.0.1:443/apis/monitoring.coreos.com/v1?timeout=32s
2020-11-05T10:17:32.888Z	INFO	controller-runtime.metrics	metrics server is starting to listen	{"addr": ":8080"}
2020-11-05T10:17:32.889Z	INFO	controllers.ScaledObject	Running on Kubernetes 1.16+	{"version": "v1.16.15-gke.3500"}
2020-11-05T10:17:32.890Z	INFO	setup	Starting manager
2020-11-05T10:17:32.890Z	INFO	setup	KEDA Version: 2.0.0
2020-11-05T10:17:32.890Z	INFO	setup	Git Commit: 50bec808f8b4656922797e4bcb7c2b61fabd6d0e
2020-11-05T10:17:32.890Z	INFO	setup	Go Version: go1.15.3
2020-11-05T10:17:32.890Z	INFO	setup	Go OS/Arch: linux/amd64
I1105 10:17:32.890565       1 leaderelection.go:242] attempting to acquire leader lease  keda/operator.keda.sh...
2020-11-05T10:17:32.890Z	INFO	controller-runtime.manager	starting metrics server	{"path": "/metrics"}
I1105 10:17:32.900857       1 leaderelection.go:252] successfully acquired lease keda/operator.keda.sh
2020-11-05T10:17:32.901Z	INFO	controller	Starting EventSource	{"reconcilerGroup": "keda.sh", "reconcilerKind": "ScaledJob", "controller": "scaledjob", "source": "kind source: /, Kind="}
2020-11-05T10:17:32.901Z	INFO	controller	Starting EventSource	{"reconcilerGroup": "keda.sh", "reconcilerKind": "ScaledObject", "controller": "scaledobject", "source": "kind source: /, Kind="}
2020-11-05T10:17:33.001Z	INFO	controller	Starting Controller	{"reconcilerGroup": "keda.sh", "reconcilerKind": "ScaledJob", "controller": "scaledjob"}
2020-11-05T10:17:33.001Z	INFO	controller	Starting workers	{"reconcilerGroup": "keda.sh", "reconcilerKind": "ScaledJob", "controller": "scaledjob", "worker count": 1}
2020-11-05T10:17:33.001Z	INFO	controller	Starting EventSource	{"reconcilerGroup": "keda.sh", "reconcilerKind": "ScaledObject", "controller": "scaledobject", "source": "kind source: /, Kind="}
2020-11-05T10:17:33.102Z	INFO	controller	Starting Controller	{"reconcilerGroup": "keda.sh", "reconcilerKind": "ScaledObject", "controller": "scaledobject"}
2020-11-05T10:17:33.102Z	INFO	controller	Starting workers	{"reconcilerGroup": "keda.sh", "reconcilerKind": "ScaledObject", "controller": "scaledobject", "worker count": 1}
```

**Comment by zroubalik** [comment]
@mangareader sorry for the delay. Have you been able to get it working? I'd recommend you that all ClusterRoles,Roles and Bindings were created in the cluster. There's nothing unusual in the logs.

**Comment by mangareader** [comment]
@zroubalik i try get curl -kv https://keda-metrics-apiserver.keda.svc.cluster.local/apis/external.metrics.k8s.io/v1beta1
``` 
{
  "kind": "APIResourceList",
  "apiVersion": "v1",
  "groupVersion": "external.metrics.k8s.io/v1beta1",
  "resources": []
}
```
the output is correct ?

**Comment by fcoedno** [comment]
I'm having the same issue here. Server version v1.16.15-gke.4300. It happens when installing either by the yaml file or with helm.

**Comment by mangareader** [comment]
@fcoedno i dont have any idea to fix the issue, i also try to upgrade to 1.17.13-gke.2001, but still get the error

**Comment by fcoedno** [comment]
Taking a look at the error message `failing or missing response from https://10.44.12.6:6443/apis/external.metrics.k8s.io/v1beta1: Get https://10.44.12.6:6443/apis/external.metrics.k8s.io/v1beta1: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)`, I've realized that this is the very same problem described in the first section of the [Troubleshooting](https://keda.sh/docs/2.0/troubleshooting/#how-can-i-use-keda-in-a-proxy-network).

That solved the problem for me @mangareader.
